### DISCUSSION TITLE
Clojure-lsp Mover para o let

### DISCUSSION THREAD

**Discussion by wandersoncferreira** [clj-br/forum#24] [Perguntas e Respostas]
É possivel usar o multiple-cursors para renomear a variavel quando eu uso `Move to let` no clojure-lsp?

Se eu tenho esse código:

```clojure
(db/query (db-fn :insert-user-data) datasource user-data)
```

e *tento* chamar a ação `Move to let` em `(db-fn ..)`, a ação não aparece como opção disponivel. **(isso eh um bug?)**

Porém se eu tenho um `let` já na função ele aparece... e resulta em:

```clojure
(let [new-binding (db-fn :insert-user-data)]
   (db/query new-binding datasource user-data))
```

Dai eu preciso chamar `rename` em `new-binding`. No clj-refactor, ele habilita o multiple-cursors e eu posso já colocar o nome da nova variavel. É possivel ter o mesmo comportamento no lsp?

**Comment by ericdallo** [comment]
> e tento chamar a ação Move to let em (db-fn ..), a ação não aparece como opção disponivel. (isso eh um bug?)

O clojure-lsp não mostra essa code action pois realmente não esta implementado para auto criar um let, mas com certeza seria um improvement a se fazer.

> Dai eu preciso chamar rename em new-binding. No clj-refactor, ele habilita o multiple-cursors e eu posso já colocar o nome da nova variavel. É possivel ter o mesmo comportamento no lsp?

Agora com o LSP 3.16, tem um novo método `window/showMessageRequest` que permite perguntar para o client alguma coisa, no caso qual seria o nome do argumento, com isso funcionaria e nao precisaria renomear apos mover :)
Para o live rename, isso só seria possivel usando uma nova feature do LSP 3.16 chamada `linked edit`, mas não esta implementada no lsp-mode ainda nem no clojure-lsp e é algo um pouco mais complexo de se fazer.

Sinta-se livre para abrir qualquer uma das issues :)
